### PR TITLE
Refactor integer min tests to use generator for tests cases

### PR DESCRIPTION
**Pull request by zoddicus**
**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [x] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Review by dneto0 — APPROVED** [review]
This is a nice cleanup!

**Comment by dneto0** [comment]
I'm not sure if there is an ordering dependency, so I'll let you land these @zoddicus

File: src/webgpu/shader/execution/builtin/min.spec.ts [hunk 1]
```diff
@@ -21,6 +21,21 @@ import { anyOf, Case, Config, correctlyRoundedThreshold, kBit, run } from './bui
 
 export const g = makeTestGroup(GPUTest);
 
+/** Generate set of min test cases from an ascending list of values */
+function generateTestCases(test_values: Array<Scalar>) {
```
Comment (ben-clayton) [inline src/webgpu/shader/execution/builtin/min.spec.ts:25]: Might be nice to include a return type on this function signature.

**Review by ben-clayton — APPROVED** [review]
Lovely stuff. One minor nit from me.

File: src/webgpu/shader/execution/builtin/min.spec.ts (hunk 1 above)
Comment (zoddicus) [reply src/webgpu/shader/execution/builtin/min.spec.ts]: Done

**Pull request merged**